src/plugins/data_importer/public/components/upload_form.tsx

```diff
@@ -0,0 +1,208 @@
+        } else if (files[0].name.endsWith('.json')) {
+          const data = JSON.parse(fileContent);
+          setPreviewData(data);
+          detectSchema(data);
+        } else if (files[0].name.endsWith('.ndjson')) {
+          const data = fileContent
+            .split('\n')
+            .filter(Boolean)
+            .map((line) => JSON.parse(line));
+          setPreviewData(data);
+          detectSchema(data);
+        }
+      };
+      reader.readAsText(files[0]);
+    }
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    // if (!file || !indexName) { // Removed cluster from the condition
+    //   setMessage('Please select a file and provide an index name');
+    //   return;
+    // }
+
+    setIsLoading(true);
+    setMessage('');
+
+    try {
+      const reader = new FileReader();
+      reader.onload = async (event) => {
+        const fileContent = event.target?.result as string;
+        let documents;
+
+        if (file?.name.endsWith('.csv')) {
+          documents = parseCSV(fileContent);
+        } else if (file?.name.endsWith('.json')) {
+          documents = JSON.parse(fileContent);
+        } else if (file?.name.endsWith('.ndjson')) {
+          documents = fileContent
+            .split('\n')
+            .filter(Boolean)
+            .map((line) => JSON.parse(line));
+        } else {
+          setMessage('Unsupported file format. Please upload a CSV, JSON, or NDJSON file.');
+          setIsLoading(false);
+          return;
+        }
+
+        const response = await http.post('/api/data_importer/upload', {
+          body: JSON.stringify({
+            documents,
+            indexName: indexName.toLowerCase(),
+            fieldMappings,
+          }),
+        });
+
+        if (response.errors) {
+          setMessage(
+            `Indexed ${response.documentsCount} out of ${response.totalDocuments} documents. ` +
+              `Failed: ${response.failedDocuments}. Check console for details.`
+          );
+          notifications.toasts.addDanger(
+            'Failed to index some documents. Check console for details.'
+          );
+        } else {
+          handleSuccess(`Successfully indexed ${response.documentsCount} documents`);
+        }
+      };
+
+      if (file) {
+        reader.readAsText(file);
+      }
+    } catch (error: any) {
+      setMessage(`Error: ${error.message || 'Unknown error occurred'}`);
+      notifications.toasts.addDanger(`Error: ${error.message || 'Unknown error occurred'}`);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const parseCSV = (csvContent: string): any[] => {
+    const lines = csvContent.split('\n');
+    const headers = lines[0].split(delimiter);
+    return lines.slice(1).map((line) => {
+      const values = line.split(delimiter);
+      return headers.reduce((obj, header, index) => {
+        obj[header] = values[index];
+        return obj;
+      }, {} as any);
+    });
+  };
```

Comment:
A library to handle this would be very useful here if we want a feature like this in the future: https://www.papaparse.com/

I have installed papaparse locally for another use case that I am working on (downloading a table into CSV), and if your work is something we want in the future, it'll be a stronger data point for us to have a library like papaparse